linkerd/http-box/src/payload.rs

```diff
@@ -0,0 +1,113 @@
+use futures::{try_ready, Poll};
+use linkerd2_error::Error;
+
+pub struct Payload {
+    inner: Box<dyn hyper::body::Payload<Data = Data, Error = Error> + Send + 'static>,
+}
+
+pub struct Data {
+    inner: Box<dyn bytes::Buf + Send + 'static>,
+}
+
+struct Inner<B: hyper::body::Payload>(B);
+
+struct NoPayload;
+
+impl Default for Payload {
+    fn default() -> Self {
+        Self {
+            inner: Box::new(NoPayload),
+        }
+    }
+}
+
+impl Payload {
+    pub fn new<B>(inner: B) -> Self
+    where
+        B: hyper::body::Payload + 'static,
+        B::Error: Into<Error>,
+    {
+        Self {
+            inner: Box::new(Inner(inner)),
+        }
+    }
+}
+
+impl hyper::body::Payload for Payload {
```

Comment:
May wish to override `content_length` as well, though for forwarded messages, the original headers take priority. :shrug: